src/index.js

```diff
@@ -54,226 +60,266 @@ function buildNametoIdMap(objectArr, type) {
-        core.info(`updated library: ${lib.name}`);
-      } else {
-        // create a new library
-        res = await createLibrary(
-          lib.name,
-          lib.description,
-          code,
-          lib.language
-        );
-        core.info(`created library: ${lib.name}`);
-      }
-      libraryDict[res.data.versionId] = { ...lib, id: res.data.id };
+// Create or update a transformation.
+async function createOrUpdateTransformation(
+  transformations,
+  transformationDict
+) {
+  core.info(`Transformations create/update started!`);
+  for (let i = 0; i < transformations.length; i++) {
+    let tr = transformations[i];
+    let code = fs.readFileSync(tr.file, "utf-8");
+    let res;
+    if (transformationNameToId[tr.name]) {
+      // update existing transformer and get a new versionId
+      let id = transformationNameToId[tr.name];
+      res = await updateTransformer(id, tr.description, code, tr.language);
+      core.info(`Updated transformation: ${tr.name}`);
+    } else {
+      // create new transformer
+      res = await createTransformer(tr.name, tr.description, code, tr.language);
+      core.info(`Created transformation: ${tr.name}`);
     }
-    core.info("Libraries create/update done!");
-
-    let transformationTest = [];
-    let librariesTest = [];
-
-    core.info("Building test suite...");
-    for (let i = 0; i < Object.keys(transformationDict).length; i++) {
-      let trVersionId = Object.keys(transformationDict)[i];
-      let testInputPath =
-        transformationDict[trVersionId]["test-input-file"] || "";
-      let testInput = testInputPath
-        ? JSON.parse(fs.readFileSync(testInputPath))
-        : "";
-      if (testInput) {
-        transformationTest.push({ versionId: trVersionId, testInput });
-      } else {
-        core.info(
-          `No test input provided. Testing ${transformationDict[trVersionId].name} with default payload`
-        );
-        transformationTest.push({ versionId: trVersionId });
-      }
+    transformationDict[res.data.versionId] = { ...tr, id: res.data.id };
+  }
+  core.info("Transformations create/update done!");
+}
+
+// Create or update a library.
+async function createOrUpdateLibrary(libraries, libraryDict) {
+  core.info(`Transformations create/update started!`);
+  for (let i = 0; i < libraries.length; i++) {
+    let lib = libraries[i];
+    let code = fs.readFileSync(lib.file, "utf-8");
+    let res;
+    if (libraryNameToId[lib.name]) {
+      // update library and get a new versionId
+      let id = libraryNameToId[lib.name];
+      res = await updateLibrary(id, lib.description, code, lib.language);
+      core.info(`Updated library: ${lib.name}`);
+    } else {
+      // create a new library
+      res = await createLibrary(lib.name, lib.description, code, lib.language);
+      core.info(`Created library: ${lib.name}`);
     }
+    libraryDict[res.data.versionId] = { ...lib, id: res.data.id };
+  }
+  core.info("Libraries create/update done!");
```

Comment:
Let's not put excessive logs. Usually logs at the starting of function is enough. Wdyt ?